### PR TITLE
Make `MODULE_NAME_PATTERN` more flexible

### PR DESCRIPTION
**Pull request by raybrownco**
Right now, format-message looks for itself being imported via the following patterns:

* `import t from 'format-message'`
* `import t from 'foo/format-message'`

In other words, the import must match `'format-message'` exactly or be prepended by a slash. This commit expands upon this to also allow imports that match these patterns:

* `import t from 'foo-format-message'`
* `import t from 'foo.format-message'`

In other words, the character that prepends `format-message` can now be a dash or a period.

This gives format-message consumers more flexibility with naming/importing their own libraries that wrap format-message. (See also: https://github.com/format-message/format-message/issues/232)

### CODE REVIEW

File: packages/format-message-estree-util/index.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 'use strict'
 
-var MODULE_NAME_PATTERN = /(^|\/)format-message$/
+var MODULE_NAME_PATTERN = /(^|\/|\.|-)format-message$/
```
Comment (vanwagonet) [inline packages/format-message-estree-util/index.js:3]: At this point, seems like we could simplify by just doing `/\bformat-message$/`

File: azure-pipelines.yml [hunk 2]
```diff
@@ -4,13 +4,17 @@
 # https://docs.microsoft.com/azure/devops/pipelines/languages/javascript
 
 pool:
-  vmImage: 'Ubuntu 16.04'
+  vmImage: 'ubuntu-latest'
```
Comment (vanwagonet) [inline azure-pipelines.yml:7]: Hmm, my neglect of this library is quite glaring.

Comment (raybrownco) [reply azure-pipelines.yml:7]: I know the feeling. I once created a Sass tool when I first joined GitHub, but didn't realize I had incoming PRs on it until four years later. 🙃

File: packages/format-message-estree-util/index.js (hunk 1 above)
Comment (raybrownco) [reply packages/format-message-estree-util/index.js:3]: Good catch!

**Comment by raybrownco** [comment]
@vanwagonet I'm running into test failures related to the `format-message-interpret` package and I'm not sure how to move forward. 

The [single test](https://github.com/format-message/format-message/blob/d3641b709052040a8264a8762b92a90d565ff52a/packages/format-message-interpret/__tests__/index.spec.js#L55) that [fails in Azure Pipelines](https://dev.azure.com/andy0862/format-message/_build/results?buildId=637&view=logs&j=b8865196-746b-550b-1739-d024186ee81d&t=83b2e335-b990-54e7-31d4-5f00f8858962&l=39) has to do with comparing zero against negative zero. The test fails in Node 10, but passes in 12, 14, and 16. This made me suspect that older versions of Node treated `format-message-interpret`'s underlying `Intl.NumberFormat()` method differently. A quick look at [MDN's Browser Compatibility table](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat#browser_compatibility) for this method shows that Node indeed didn't add full support for it until Node 12/13.

I feel like I've got two pragmatic outcomes here:

1. Remove the `-0 should output -0` test
2. Remove Node 10 from the pipeline matrix since it's already reached End-of-Life

Which option would you go with?

**Comment by vanwagonet** [comment]
@raybrownco dropping support for obsolete Node versions sounds preferable to me.

**Comment by raybrownco** [comment]
@vanwagonet Done and done. This PR is ready for your review and approval. If you see anything else that needs changing, please just let me know. Thanks so much!

**Review by vanwagonet — APPROVED** [review]
